casa_formats_io/casa_low_level_io.py

```diff
@@ -820,6 +929,101 @@ def read_header(self, f):
 
         super().read_header(f)
 
+    def read_column(self, filename, seqnr, column, coldesc, colindex_in_dm):
+
+        # Need to expose the following somehow
+        target_chunksize = None
+        memmap = False
+
+        # chunkshape defines how the chunks (array subsets) are written to disk
+        chunkshape = tuple(self.default_tile_shape)
+        chunksize = np.product(chunkshape)
+
+        # the total shape defines the final output array shape
+        totalshape = self.cube_shape
+
+        # we expect that the total size of the array will be determined by finding
+        # the number of chunks along each dimension rounded up
+        totalsize = np.product(np.ceil(totalshape / chunkshape)) * chunksize
+
+        # the ratio between these tells you how many chunks must be combined
+        # to create a final stack
+        stacks = np.ceil(totalshape / chunkshape).astype(int)
+        nchunks = int(np.product(stacks))
+
+        dtype = TO_DTYPE[coldesc.value_type]
+
+        if coldesc.value_type != 'bool':
+            if self.big_endian:
+                dtype = '>' + dtype
+            else:
+                dtype = '<' + dtype
+
+        itemsize = np.dtype(dtype).itemsize
+
+        chunkshape = tuple(int(x) for x in chunkshape)
+        totalshape = tuple(int(x) for x in totalshape)
+
+        # CASA chunks are typically too small to be efficient, so we use a larger
+        # chunk size for dask and then tell CASAArrayWrapper about both the native
+        # and target chunk size.
+        # chunkshape = determine_optimal_chunkshape(totalshape, chunkshape)
+
+        if target_chunksize is None:
+            target_chunksize = 10000000
+
+        if chunksize < target_chunksize:
+
+            # Find optimal chunk - since we want to be efficient we want the new
+            # chunks to be contiguous on disk so we first try and increase the
+            # chunk size in x, then y, etc.
+
+            chunkoversample = previous_chunkoversample = [1 for i in range(len(chunkshape))]
+
+            finished = False
+            for dim in range(len(chunkshape)):
+                factors = [f for f in range(stacks[dim] + 1) if stacks[dim] % f == 0]
+                for factor in factors:
+                    chunkoversample[dim] = factor
+                    if np.product(chunkoversample) * chunksize > target_chunksize:
+                        chunkoversample = previous_chunkoversample
+                        finished = True
+                        break
+                    previous_chunkoversample = chunkoversample
```

Comment:
Since `chunkoversample` and `previous_chunkoversample` are the same object, I think the logic here doesn't work.  There needs to be a copy operation somewhere.  Right?

Reply:
Good catch!